src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,526 @@
+                        sortable: "string",
+                        extraClass: (x) => x.effectiveValue() ? "value-has-content" : ""
+                    }),
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValueOrigin(), "Origin", "20%", {
+                        sortable: "string",
+                        extraClass: (x) => {
+                            let classes = "source-item ";
+                            switch (x.effectiveValueOrigin()) {
+                                case "Database":
+                                    classes += "source-item-database";
+                                    break;
+                                case "Server":
+                                    classes += "source-item-server";
+                                    break;
+                                case "Default":
+                                    classes += "source-item-default";
+                                    break;
+                            }
+                            return x.effectiveValue() ? classes : classes += " null-value";
+                        }
+                    })
+                ]
+            }
+        });
+    }
+    
+    private fetchSummaryData(): JQueryPromise<pagedResult<models.settingsEntry>> {
+        let entriesForSummaryMode = _.filter(this.allEntries(), x => x.showEntry());
+
+        entriesForSummaryMode = entriesForSummaryMode.reduce((acc,element) => {
+            if (element.entryClassForSummaryMode() === "highlight-key") {
+                return [element, ...acc];
+            }
+            return [...acc, element];
+        }, []);
+        
+        return $.when<pagedResult<models.settingsEntry>>({
+            items: entriesForSummaryMode,
+            totalResultCount: entriesForSummaryMode.length,
+            resultEtag: null,
+            additionalResultInfo: undefined
+        })
+    }
+
+    private fetchDatabaseSettings(db: database, refresh: boolean = false): JQueryPromise<Raven.Server.Config.SettingsResult> {
+        eventsCollector.default.reportEvent("database-settings", "get");
+
+        return new getDatabaseSettingsCommand(db, refresh)
+            .execute()
+            .done((result: Raven.Server.Config.SettingsResult) => {
+                
+                const settingsEntries = result.Settings.map(x => {
+                    const rawEntry = x as Raven.Server.Config.ConfigurationEntryDatabaseValue;
+                    return models.settingsEntry.getEntry(rawEntry);
+                });
+
+                this.allEntries(_.sortBy(settingsEntries, x => x.keyName()));
+            });
+    }
+    
+    private findFirstCategoryWithError() {
+        let categoryWithError: string;
+        
+        this.allEntries().some(entry => {
+            if (this.getEntryStateClass(entry) === "invalid-item") {
+                return categoryWithError = entry.data.Metadata.Category;
+            }
```

Comment:
please also keep in mind this function has no explicit return

Reply:
ok